In [73]:
import seaborn as sns
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
import plotly.express as px
import plotly.io as pio
import plotly.graph_objects as go
from plotly.subplots import make_subplots

In [74]:
df_sales_ind = pd.read_csv('sales_ind_clean.csv', index_col=False, parse_dates=["release_date"])
df_sales_series = pd.read_csv('sales_series_clean.csv', index_col=False, parse_dates=["release_date"])

sns.set_style("whitegrid")
pio.templates.default = "seaborn"

df_sales_ind = df_sales_ind.sort_values('units_reported', ascending=False)

In [75]:
# 1 - Evolución del mercado a lo largo del tiempo

# ¿Cómo han evolucionado las ventas de los videojuegos a lo largo de los años?


# Variables total de datos
sales_per_year = df_sales_ind.groupby('year').agg(
    sales=('units_reported', 'sum'),
    games_released=('units_reported', 'count'),
    sales_mean = ('units_reported', 'mean'),
    sales_median = ('units_reported', 'median')
).reset_index().copy()

df_long_year = sales_per_year.melt(
    id_vars=["year"],
    value_vars=['sales_mean', 'sales_median'],
    var_name="statistic",
    value_name="value"
)

# Variables 10% de datos
sales_top_10_percent = df_sales_ind.iloc[:int(len(df_sales_ind) * 0.10)]

top_grouped = sales_top_10_percent.groupby('year').agg(
    sales=('units_reported', 'sum'),
    games_released=('units_reported', 'count'),
    sales_mean = ('units_reported', 'mean'),
    sales_median = ('units_reported', 'median')
).reset_index().copy()

df_long_10 = top_grouped.melt(
    id_vars=["year"],
    value_vars=['sales_mean', 'sales_median'],
    var_name="statistic",
    value_name="value"
)

# Gráfico ventas totales por año de origen

fig = px.line(sales_per_year, x="year", y="sales", markers=True,
              title='Ventas totales según año de lanzamiento',
              hover_data=['games_released'])

fig.update_layout(
    xaxis_title="Año de lanzamiento",
    yaxis_title="Ventas totales (millones de copias)"
)

fig.update_traces(line_color="crimson")

fig.show()


fig = px.line(top_grouped, x="year", y="sales", markers=True,
              title='Ventas exitosas según año de lanzamiento',
              hover_data=['games_released'])

fig.update_layout(
    xaxis_title="Año de lanzamiento",
    yaxis_title="Ventas totales (millones de copias)"
)

fig.update_traces(line_color="crimson")

fig.show()


# Gráfico promedio y mediana de ventas por juego según año de origen

fig = px.line(df_long_year, x="year", y="value", markers=True, color='statistic',
              title='Promedio y Mediana según año de lanzamiento')

fig.update_layout(
    xaxis_title="Año de lanzamiento",
    yaxis_title="Promedio/Mediana de ventas"
)

fig.show()

fig = px.line(df_long_10, x="year", y="value", markers=True, color='statistic',
              title='Promedio y Mediana según año de lanzamiento (10%)')

fig.update_layout(
    xaxis_title="Año de lanzamiento",
    yaxis_title="Promedio/Mediana de ventas"
)

fig.show()

# Gráfico cantidad de juegos según año de origen

fig = px.histogram(df_sales_ind, x="year", color_discrete_sequence=['#00CC96'] )

fig.update_layout(
    xaxis_title="Año de lanzamiento",
    yaxis_title="Cantidad de juegos"
)

fig.update_traces(
    marker_line_color='black',  
    marker_line_width=0.5      
)

fig.show()


fig = px.histogram(sales_top_10_percent, x="year", color_discrete_sequence=['#00CC96'] )

fig.update_layout(
    xaxis_title="Año de lanzamiento",
    yaxis_title="Cantidad de juegos"
)

fig.update_traces(
    marker_line_color='black',  
    marker_line_width=0.5       
)

fig.show()

## Interpretación

### Ventas totales por año de origen
En el primer gráfico, se pueden observar las unidades totales vendidas según el año de origen de cada juego. Es decir, el pico de 1322 millones de ventas del año 2011 en realidad nos dice que los juegos que se originaron ese año acumulan la mayor cantidad de ventas del conjunto de datos estudiado.  
  
Este gráfico, sin embargo, no nos habla del éxito de la industria de los videojuegos en cada año, ya que un juego originado en 2011 no necesariamente fue vendido ese mismo año. Aun así, estos datos nos pueden decir información importante de que tan activa ha estado la industria, y nos ayuda a comprender mejor la naturaleza de los datos.

Los resultados de la primera imagen nos dicen que los videojuegos originados entre los años 2005 y 2018 son los que más han acumulado ventas, destacando especialmente los años 2011 y 2008, pero ¿Significa esto que los juegos más vendidos provienen de esos dos años?  

Es válido hacerse esa pregunta, ya que el primer gráfico no tiene en cuenta la cantidad de juegos que fueron lanzados cada año.  

El segundo gráfico nos muestra las ventas totales, pero teniendo en cuenta el 10% de videojuegos más vendidos del conjunto de datos. Se puede observar que la tendencia de la línea es batante parecida a la del gráfico que usa todos los datos, sugiriendo que la cantidad de videojuegos exitosos de cada año de origen es proporcional a sus ventas totales y videojuegos producidos. Al calcular algunas de estas proporciones, se concluye que alrededor del 8% de los videojuegos lanzados cada año terminaron siendo exitosos, con la excepción de algunos años importantes como 2009, que tuvo menos éxitos.

### Promedio y mediana de ventas por año de origen
En el tercer gráfico, que corresponde al promedio y mediana de los datos completos, vemos que la línea sigue una tendencia diferente al de las ventas totales: en los años donde el primer gráfico exhibía puntos álgidos de ventas acumuladas, se puede observar una línea estable sin mucha variación en los gráficos de media y mediana. Por el contrario, podemos observar que el promedio de ventas de los años 1985, 2021 y 2023 muestran valores bastante altos. En el caso de la mediana, la línea sigue la misma tendencia que el promedio, pero de manera menos extrema, lo cual nos indica que para casi todos los años de origen, unos pocos grandes éxitos están inflando el promedio.  

Sin embargo, el hecho de que tanto el promedio como la mediana muestren valores altos en lugares donde las ventas totales no parecen mostrar nada especial (y viceversa), nos podría estar sugiriendo que la cantidad de juegos que se lanzaron cada año podrían tener que ver con estos resultados, y valdría la pena analizarlas.

En el cuarto gráfico, que corresponde al promedio y mediana del 10% de videojuegos más vendidos, se puede observar algo diferente: la línea adquiere una tendencia más parecida a la del gráfico de ventas totales para algunos años, como el segmento entre los años 2006 y 2018. Este suceso es probablemente causado porque al tomar el 10% de los datos, aislamos los videojuegos con alta cantidad de ventas, incluyendo los que tuvieron un enorme éxito y por lo tanto sesgan los resultados, de una forma similar al caso de las ventas totales. La cantidad abundante de años de origen que parecen alzarse en resultados de promedio o mediana al tomar el 10% de los datos sugiere que dos de las ideas previas que habían surgido son probablemente ciertas: que el éxito de las ventas está sesgado por unos cuantos éxitos, y que han salido juegos exitososos en la mayoría de años incluidos en el conjunto de datos. Esto último se evidencia por otra diferencia clave entre los dos gráficos de promedio/mediana: La mayoría de años de origen exhiben puntos altos en el gráfico del 10% que no tenían en el gráfico de los datos completos.

### Cantidad de juegos lanzados por año
En el quinto gráfico de esta parte del análisis, que utiliza los datos completos, podemos observar algo importante: la mayor cantidad de videojuegos producidos se concentra entre los años 2000 y 2018, una tendencia que se ve similar a lo que se observa en el gráfico de las ventas totales. Esto confirma las sospecha que había surgido al analizar los anteriores gráficos: la cantidad de juegos producidos por año tiene directa relación con la dicotomía entre la tendencia de la línea de ventas totales y la de las líneas de promedio y mediana.  

Gracias a esto podemos concluir que la razón por la que años como 1985, 2021 y 2023 exhiben altos valores de mediana y promedio se debe a que, incluso siendo años donde se lanzaron pocos juegos (en comparación a otros años), gran parte de ellos tuvieron una considerable cantidad de ventas.  

Otro importante factor a destacar es el innegable hecho de que, según nuestro conjunto de datos, desde el año 202 en adelante se han lanzado menos juegos que en los 20 años anteriores, donde se concentra la producción de juegos. A pesar de esto, si vemos el gráfico de ventas totales, nos daremos cuenta que las ventas de los juegos que se originaron después del año 2020 mantienen el nivel general de las ventas entre 2000 y 2020, si obviamos que no superan grandes éxitos como los años 2008, 2011 y 2017. Esta información es útil, ya que nos dice que en los años más recientes, los videojugos han seguido teniendo éxito a pesar de haberse producido menos, esto último pudiendo deberse a falta de datos o a una realidad de la industria.  

Cuando se analiza el sexto y último gráfico, que muestra la cantidad de juegos lanzados por año en el 10% de juegos más vendidos, se observan leves diferencias. Para empezar, los años posteriores a 2018 muestran barras más largas que en el gráfico anterior, lo cual sugiere que dichos años tienen un mayor porcentaje de lanzamientos exitosos que la mayoría de años, característica que comparten en menor medida los años previos a 1994. Este logro se puede atribuir, por supuesto, a la baja cantidad de juegos lanzados en total de estos años, pero el hallazgo sigue siendo digno de investigación. Por lo demás, la distribución de los datos es similar a la del gráfico anterior, con pequeñas excepciones de años que tienen una proporción de éxitos más baja como 2009.

### Conclusiones
Con esta información, se pueden sacar varias cosas en claro del conjunto de datos:  

-Existe una gran diferencia de datos entre los años de orgien de lanzamiento de los juegos, dejando claro que las medidas de ventas acumuladas, promedio y mediana se ven afectadas por esta desigualdad. Sin embargo, no se deben ignorar estos resultados, ya que, aunque no representen directamente el éxito financiero de la industria durante esos años, es importante destacar que los videojuegos lanzados desde el año 2019 en adelante tuvieron un mejor desempeño general que los que se lanzaron el resto de años, algo que es destacable incluso con la menor cantidad de juegos producidos por año. 
Algunos años a destacar por este tipo de resultados serían:
+ El año 2019, con un promedio de 8.75 y una mediana de 5.5 millones de copias vendidas de 47 éxitos lanzados.
+ El año 2023, con un promedio de 9.2 y una mediana de 4.5 millones de copias vendidas de 26 éxitos lanzados.

Es importante poner atención a estos años, no solo porque tienen estadísticas de venta interesantes, si no porque representan los años más actuales del conjunto de datos, lo cual hace importante su información si nuestro objetivo fuera tener éxito en la industria actual de videojuegos.  

-Años de origen de videojuegos que destacan por su cantidad de ventas acumuladas también merecen atención: incluso considerando que su éxito al medir dicha estadística se deba en parte a la numerosa cantidad de juegos lanzados esos años, existen determinados casos en los que hay otras razones:
+ El año 2011 lidera las ventas acumuladas con 1021 millones de copias de 113 juegos exitosos.
+ El año 2008 destaca con 520 millones de copias de 127 juegos exitosos.
+ El año 2017, con 760 millones de copias de 76 juegos exitosos.

Una razón importante por la que vale la pensa investigar estos años es que, al comparar el gráfico de ventas totales y el de cantidad de juegos lanzados por año, nos damos cuenta que el éxito de estos años de origen no se debe únicamente a la cantidad de juegos, ya que otros años con lanzamientos similares muestran una estadística de venta total mucho menor, como es el caso del año 2009, el año con más juegos lanzados.

Será la responsabilidad de los estudios de las siguientes partes del análisis determinar la naturaleza del éxito de estos años de origen, y averiguar si se debe a resultados favorables aislados o si en efecto se demuestra que tienen un buen rendimiento general. 
Es importante recalcar que esta información no nos lleva directamente a responder nuestra pregunta inicial sobre los factores que implican éxito en la industria, pero si nos acerca entregandonos información importante.





In [76]:
# 1.5 - Estudio menor: Variación de ventas por región en diferentes años de origen

df_sales_region = df_sales_ind[df_sales_ind['total_sales'].notna()]

sales_per_region = df_sales_region.groupby('year').agg(
    sales=('total_sales', 'sum'),
    games_released=('total_sales', 'count'),
    na_sales=('na_sales', 'sum'),
    jp_sales=('jp_sales', 'sum'),
    pal_sales=('pal_sales', 'sum'),
    other_sales=('other_sales', 'sum'),
).reset_index().copy()
sales_per_region = sales_per_region[sales_per_region['year'].between(1998, 2018)]

regional_sales = ['na_sales', 'pal_sales', 'other_sales', 'jp_sales']
important_years = [2008, 2011, 2017]

df_long_region = df_sales_region.melt(
    id_vars=["year", "title"],
    value_vars=regional_sales,
    var_name="region",
    value_name="region_sales"
)

df_long_region = df_long_region.sort_values("region_sales", ascending=False)

# print(df_long_region)

fig = px.bar(sales_per_region, x='year', y=regional_sales)

fig.update_layout(
    title="Ventas por región (año de origen)",
    xaxis_title="Año de lanzamiento",
    yaxis_title="Ventas"
)

fig.show()



for year in important_years:
    df_long_year = df_long_region[df_long_region['year'] == year].copy()
    df_long_year = df_long_year[df_long_year['region_sales'] > 0]
    df_long_year["ranking"] = range(1, len(df_long_year) + 1)
    fig = px.scatter(
    df_long_year,
    x="ranking",
    y="region_sales",
    color="region",
    hover_name="title"
    )
    fig.update_layout(
    title=f"Mayores ventas de juegos año {year}",
    xaxis_title="Ranking",
    yaxis_title="Ventas"
    )
    fig.show()    




## Ventas por región

Antes de hablar de los gráficos, vale la pena mencionar que para esta sección se trabajó con una menor cantidad de datos, específicamente un 80% de los datos limpios, debido a que no todas las filas contenían información sobre las ventas en cada región. Además, algunos de los registros que no se usan en esta sección corresponden a videojuegos con los números de ventas más altos de el conjunto de datos. Sin embargo, hacer un análisis de ventas por región sigue siendo valioso: un 80% de datos es suficiente para sacar conclusiones importantes.

## Gráfico de barras apiladas

En el primer gráfico se trabajó con datos entre los años 1998 y 2018. La razón detrás de esto es que los años previos a 2000 y posteriores a 2018 no tenían suficientes registros y/o no mostraban una relación ni una tendencia relevante de las ventas entre regiones. Además, nos permite visualizar con mayor claridad el período donde se concentran las ventas.  

Un conocimiento general que se puede extraer del gráfico es el hecho de que las ventas en cada región aumentan con el tiempo, hasta llegar a un pico entre los años 2008 y 2010. Todas las regiones siguen este patrón. En un principio, es la región de Norteamérica la que domina las ventas de años importantes como 2010, 2011 y 2012. Sin embargo, en ciertos años como 2014 o 2016 observamos que las ventas de la región PAL (incluyendo Europa, Australia y parte de África, Asia y Sudamérica) se igualan las de Norteamérica, por lo que podemos deducir que, a medida que ha pasado el tiempo, los juegos más modernos han logrado venderse equitativamente a las dos regiones con más ventas, Norteamérica y PAL, cuando la situación no era así en años anteriores.  

Japón y las regionas calificadas como "Otras" han experimentado similares subidas y bajadas a lo largo de los años, pero se diferencian en el hecho de que Japón alcanza en el año 2011, mientras que las regionas categorizadas como "Otras"  alcanzaron su máximo de ventas en el año 2008, al igual que la región de PAL y la de Norteamérica. Esto marca una de las escasas diferencias entre las ventas en Japón y el resto de regiones.

## Gráficos de ranking de años importantes

En el estudio previo se concluyó que algunos de los años importantes para estudiar eran 2008, 2011 y 2017, que además representan diferentes épocas por las que pasaba la industria, por lo cual se graficaron los juegos individuales ordenados por ventas y coloreados por región de cada uno de esos años. Esto nos permitió observar la distribución de los datos, y confirmar nuestra sospecha de que hay videojuegos con enormes cantidad de ventas que influyen considerablemente en los resultados.  

En los 3 años de origen observados, las ventas más altas ocurrieron en las regiones de Norteamérica y PAL, como era de esperarse, y los valores atípicos corresponden en su mayoría a los juegos más vendidos de cada año de origen, en sus diferentes plataformas y regiones:  
+ "Grand Theft Auto IV" en 2008.
+ "Call of Duty: Modern Warfare 3" en 2011.
+ "Call of Duty: WWII" en 2017.

Dichos títulos tienen en buena parte la responsabilidad del éxito que se observa en las estadísticas para su respectivo año de origen, y vale la pena estudiar sus características para comprender que los llevó a conseguir tantas ventas. 

## Conclusiones

Dado que las ventas regionales siguen una tendencia parecida a la de los años de origen, esto último se tendrá más enconsideración a la hora de estudiar otras características de la industria, ya que, como se mencionó anteriormente, esta parte del análisis se realizó únicamente con el 80% de los datos procesados. Sin embargo, no se puede negar que se sacaron importantes conclusiones de este pequeño estudio, como el hecho de que las ventas de Norteamérica suelen concentrarse más en años anteriores, para después igualarse más con el resto de regiones. Además, los títulos que descubrimos como responsables de la mayoría del éxito de años que ya considerábamos importantes serán una buena herramienta de análisis más adelante.   


In [77]:
# 2 - Estudio de ventas por género

sales_by_genre = df_sales_ind.groupby('genre').agg(
    sales=('units_reported', 'sum'),
    games_released=('units_reported', 'count'),
    sales_mean = ('units_reported', 'mean'),
    sales_median = ('units_reported', 'median')
).reset_index().sort_values('sales', ascending=False).copy()

df_long = sales_by_genre.melt(id_vars ='genre', value_vars=['sales_mean', 'sales_median'],
                              var_name='statistic', value_name='value')

# Gráfico ventas totales
fig = px.bar(sales_by_genre, x='genre', y='sales')

fig.update_layout(
    title="Ventas por género",
    xaxis_title="Género",
    yaxis_title="Ventas"
)
fig.update_xaxes(tickangle=60) 
fig.update_traces(marker_color='crimson')

fig.show()


# Gráfico Top 10
top_10_genres = sales_by_genre.iloc[:10]

fig = px.pie(top_10_genres, values='sales', names='genre', title='Top 10 géneros con más ventas totales')
fig.show()


# Gráfico media vs. mediana
fig = px.bar(df_long, x='genre', y='value', color='statistic', barmode='group')
fig.update_layout(
    title="Promedio vs. Mediana de ventas por género",
    xaxis_title="Género",
    yaxis_title="Promedio/Mediana"
)
fig.update_xaxes(tickangle=60) 
fig.show()


# Gráfico de cantidad de juegos por género



fig = px.histogram(df_sales_ind, x="genre", color_discrete_sequence=['#00CC96'] )
fig.update_layout(
    title="Juegos lanzados por género",
    xaxis_title="Género",
    yaxis_title="Cantidad de juegos"
)
fig.update_xaxes(tickangle=60) 
fig.show()


# Gráfico 10% 
sales_top_10_percent = df_sales_ind.iloc[:int(len(df_sales_ind) * 0.10)]

proportion_10 = sales_top_10_percent['genre'].value_counts()
proportion_full = df_sales_ind['genre'].value_counts()
genre_proportion = (proportion_10 / proportion_full).fillna(0).sort_values(ascending=False)

fig = px.bar(x=genre_proportion.index, y=genre_proportion)
fig.update_yaxes(tickformat=".0%")
fig.update_layout(
    title="Presencia de géneros en entregas exitosas",
    xaxis_title="Género",
    yaxis_title="% en mejores vendidos"
)
fig.update_xaxes(tickangle=60) 
fig.show()


# Gráfico de juegos más vendidos de cada género

important_genres = ["Action-Adventure", "Shooter", "Role-Playing", "Sports", "Party", "Platform"]
top_10_unique = sales_top_10_percent.groupby(['genre']).head(2).copy()
top_10_unique = top_10_unique[top_10_unique['genre'].isin(important_genres)]

fig = px.scatter(top_10_unique, y="genre", x="units_reported", color="title")
fig.update_layout(
    title="Títulos más exitosos por género",
    xaxis_title="Ventas",
    yaxis_title="Géneros"
)
fig.update_traces(marker_size=15)
fig.show()

## Ventas por género

En los primeros dos gráficos de esta sección exploramos las ventas totales separadas por el género de los videojuegos. Podemos observar que los tres géneros más prominentes en ventas, bastante cerca unos de otros, son los juegos de Acción-Aventura, de Disparos y de Rol, los tres superando los dos millones de ventas. Les siguen géneros relevantes pero menos exitosos: Deportes, Acción y Plataformas. En el segundo gráfico podemos observar los porcentajes exactos de las ventas de cada uno de los diez géneros con más ventas. La equitativa división de las ventas en este gráfico circular nos dice que no existe un único género que acapare las ventas de los videojuegos, si no que varios de estos parecen tener éxito en la industria. Sin embargo, sin incluir más elementos en el estudio, no podemos comprender aún la razón detrás de los resultados que vemos hasta ahora.

## Promedio y mediana de ventas por género

En este gráfico, que compara el promedio y la mediana de las ventas de cada género, observamos resultados diferentes a los de los gráficos aneriores: Promedos altos y medianas bajas, con un orden de tamaños diferente al de las ventas totales. Sin embargo, esto podría deberse a la naturaleza asimétrica de los datos, donde existen unos pocos valores muy altos que influencian los resultados. Esto es inevitable y una característica clave del entorno que se está estudiando.  

En este caso, se puede observar que Acción-Aventura sigue siendo el género dominante, superando a todo el resto de géneros en mediana y sobretodo en promedio, lo cual podría indicarnos que existen videojuegos concretos de este género que tuvieron un gran éxito en la industria, lo cual lo convierte en una variable de interés. Géneros como Disparos y Rol mantienen medidas similares, pero esta vez el género de Plataformas muestra mejores estadísticas que Acción y Deportes, lo cual era diferente para las ventas totales.  

Existen dos géneros que marcan una diferencia más grande con el gráfico de ventas totales: Juegos de fiesta y MMO (Multijugador masivo). Ambos exhiben una mediana y promedio mayor a la de la mayoría de géneros, en especial los juegos de fiesta. Este hecho es digno de estudiar, porque la diferencia entre el promedio/mediana de estos géneros con el de sus ventas totales es inmensa, y pasan de ser de los géneros menos vendidos a tener de las estadísticas más altas. Sin embargo, para comprender se debe hacer un paso similar al que se hizo en el estudio de los años de origen: Averiguar la cantidad de videojuegos producidos para cada género.

## Cantidad de juegos por género

Gracias a la visualización de este cuarto gráfico, nos damos cuenta de varias cosas. Primero, que el género de Acción-Aventura, sorprendentemente, es uno de los géneros con menos juegos lanzados, lo cual hace su cantidad de ventas totales un logro aún mayor del que ya se consideraba. El hecho de que el gpenero de Acción-Aventura tenga tan pocos títulos podría deberse a que también existe el género Acción, que se alza como el género con mayor cantidad de juegos lanzados, y que podría ser parte de juegos similares a los de Acción-Aventura. Sin embargo, la diferencia de números en ventas del género compuesto nos sugiere que existe una diferencia: El género de Acción vende más cuando está combinado con Aventura. Esto se hace más evidente cuando comprobamos que el género Aventura, por sí solo, no muestra estadísticas del todo interesantes.  

Otros géneros populares con muchos juegos lanzados son Deportes y Rol, lo cual los consolida como géneros de interés por sus notables resultados en los gráficos visualiazdos. Algo similar sucede con los géneros Disparos y Plataformas, géneros populares en ventas pero menos producidos que los anteriormente mencionados.  

Por último, se tiene el caso de los juegos de fiesta y MMO. Están entre los 5 géneros con menos juegos lanzados, lo cual explicaría en parte sus altas cifras en en el gráfico de promedio y mediana. Sin embargo, este caso no se repetía para géneros como Música o Novelas Visuales, que, a pesar de haber pocos títulos que pertenezcan a estos géneros, sus promedios, medianas y ventas totales no muestran nada destacable, por lo que el caso de los géneros MMO y de fiesta es digno de mayor investigación.

## Procentaje de géneros presentes en los títulos mejor vendidos

Para elaborar este gráfico, se tomó el decil superior de las ventas, analizando que porcentaje del total de los videojuegos de cada género estaban presentes en el top 10%.  

Como de costumbre, el primer género en aparecer es Acción-Aventura, con un 36% de sus juegos en el top de ventas. Esto no es una sorpresa, ya que los análisis anteriores habían probado que este género es un éxito total, con 226 juegos en el top de ventas.  

Al primer puesto le siguen los géneros de fiesta y MMO, que ya han sido destacados por sus altos promedios y medianas en ventas. De acuerdo a los porcentajes, existen 16 juegos de fiesta y 11 juegos MMO en el top de ventas, lo cual parece poco, pero es una buena cantidad considerando que el total de juegos de ambos géneros son 54 y 49 títulos respectivamente.  

Luego de esto, siguen dos géneros a los que se les ha puesto interés en anteriores análisis, Disparos y Plataformas, con 280 y 188 juegos en el top de ventas, respectivamente. Ligeramente por detrás, el género de Deportes tiene un 10% de sus juegos en el top de ventas, sin embargo, debido a la alta cifra de videojuegos con este género, un 10% sigue siendo bastante: corresponde a 266 juegos.

Estos últimos resultados consolidan los géneros estudiados como géneros de interés, debido a las constantes ocasiones en que muestran resultados positivos. Esto incluye al género de fiesta y MMO, que a pesar de tener una cantidad menor de juegos en el top de ventas, su constancia en tener éxito en una considerable parte de los juegos lanzados es notable. Dicho esto, existen otros géneros con resultados decentes, como lo son los juegos de Lucha o Carreras, pero se decidió no incluirlos en el siguiente análisis como géneros de interés para concentrarse en los otros seis géneros que se han mencionado en las conclusiones anteriores.

## Títulos exitosos por género y conclusiones

En el último gráfico de la sección podemos observar los dos títulos más vendidos de cada género de interés, junto a la cantidad de ventas que generó cada uno. Además , se realizó una investigación de cada título presente en el gráfico, con el af´na de comprender la naturaleza de su éxito.  

Lo primero que se puede notar es que, como ha sido el caso en el resto de los gráficos, los resultados del género Acción-Aventura sobrepasan a los demás por una inmensa cantidad de ventas, con los videojuegos "Minecraft" y "Grand Theft Auto V", dos videojuegos conocidos por su libertad de juego, capacidad de adaptarse a todo tipo de jugador y a su profundidad, al ofrecer al jugador muchas opciones de contenido que pueden explorar. Además, son juegos que permiten a los usuarios utilizar servidores para jugar con sus amigos o con gente de todo el mundo. 

En el caso de Deportes, un género que ha tenido resultados positivos en nuestros análisis previos, tenemos las entregas "Wii Sports" y "Wii Sports Resort", dos entregas que revolucionaron la industria al usar controles de movimiento intuitivo, y se volvieron populares al venderse junto a la consola donde se juegan, la Wii.  

Para los videojuegos de Disparos, dos videojuegos aparecen como los más vendidos por el popular modod de juego "Battle Royale", siendo el videojuego "Player Unknown Battlegrounds" el que popularizó esta forma de jugar, mientras que "ArmA II" es un caso curioso, ya que debe gran parte de su éxito a "DayZ", una modificación del juego base que utiliza el modo de juego "Battle Royale". Por esto, podemos asociar buena parte del éxito de los juegos de disparos al fenómeno "Battle Royale", sin embargo, hay muchos juegos de disparos que vendieron una cantidad decente de unidades pero que no están relacionados al modo de juego mencionado, como es el caso de "Call of Duty: Modern Warfare 3", un videojuego que mencionamos en un estudio anterior.  

Para el género de plataformas, tenemos dos casos distintos: El de "Terraria", un juego considerado un "clásico de nicho" que combina elementos de plataformas con Acción, Aventura y Sandbox de forma similar a "Minecraft". A pesar de ser de los juegos más vendidos, la percepción popular de "Terraria" es que no es conocido por muchos. Luego está el caso del segundo juego más vendido del género, "Human Fall Flat", que se hizo popular por su enfoque cooperativo y el impacto que generaron creadores de contenido en plataformas como YouTube y Twitch.  

Por último, "The Witcher 3: Wild Hunt" y "The Elders Scrolls V: Skyrim", los videojuegos más vendidos de Rol, destacan por su libertad de exploración y mundo inmersivo de fantasía, y en el caso de Skyrim, por una comunidad de jugadores que hacen mods (modificaciones) del juego, manteniéndolo vivo incluso más de una década después de su lanzamiento.  

En conclusión, este estudio nos ha dejado claro cuáles géneros han tenido mayor éxito en la industria, y de que forma lo han logrado. Además de seleccionar varios géneros de interés, este estudio nos permitió explorar diferentes títulos de gran éxito para comprender lo que causó que fueran tan vendidos. Por lo demás, este estudio nos permitió comprender aún mejor el conjunto de datos elegido para el estudio y nos dejó en claro que cuando se trata de géneros y tipos de juegos, no hay una única respuesta para llegar al éxito, aunque sí hay factores clave en los que se debe hacer foco, como la adaptabilidad, libertad, o la innovación que puedan tener los videojuegos.




In [78]:
#  2.5 - Inclusión de años y regiones en el estudio de géneros

pd.set_option('display.max_rows', None)

# Gráfico subplot de géneros

genre_by_year = df_sales_ind.groupby(
    ["year", "genre"],
    as_index=False
).agg({
    "units_reported": "sum",
}).sort_values('units_reported', ascending=False)

important_years = [2008, 2011, 2019, 2023]

fig = make_subplots(
    rows=2,
    cols=2,
    specs=[
        [{"type": "pie"}, {"type": "pie"}],
        [{"type": "pie"}, {"type": "pie"}]
    ],
    subplot_titles=[
        "Géneros más vendidos 2008", 
        "Géneros más vendidos 2011",
        "Géneros más vendidos 2019",
        "Géneros más vendidos 2023"]
)

for i, year in enumerate(important_years):
    df_year = genre_by_year[genre_by_year['year'] == year].head(10).copy()
    fig.add_trace(
        go.Pie(labels=df_year["genre"], values=df_year["units_reported"]),
        row = i // 2 + 1,
        col = i % 2 + 1
    )
    fig.update_layout(
    width=1000,
    height=800
    )

for annotation in fig.layout.annotations:
    annotation.y += 0.04

fig.show()

# Gráfico género por regiones

genre_per_region = df_sales_region.groupby('genre').agg(
    sales=('total_sales', 'sum'),
    na_sales=('na_sales', 'sum'),
    jp_sales=('jp_sales', 'sum'),
    pal_sales=('pal_sales', 'sum'),
    other_sales=('other_sales', 'sum'),
).reset_index().sort_values('sales', ascending=False).copy()


fig = px.bar(genre_per_region, x='genre', y=regional_sales)

fig.update_layout(
    title="Ventas por género",
    xaxis_title="Género",
    yaxis_title="Ventas"
)
fig.update_xaxes(tickangle=60) 

fig.show()

### Géneros a través de las regiones y años de origen  

En los últimos dos gráficos, se hizo un pequeño estudio adicional involucrado a los géneros, pero esta vez relacionándolos con el año de origen y region de ventas de los videojuegos.  

La primera visualización se compone de cuatro gráficos circulares que representan años de origen que fueron determinados como importantes en un análisis previo. Ayudan a ver qué géneros corresponden a los juegos más populares que salieron en esos años, así como demostrar como han cambiado la producción y popularidad de juegos de diferentes géneros. De acuerdo a los resultados, tanto para el año 2011, 2019 y 2023, el género más popular fue Acción-Aventura, con un porcentaje de ventas mayor al 25%. Esto, sumado al hecho de que aparece en los cuatro gráficos, apoya la idea de que el género ha sido popular históricamente y su enorme cantidad de ventas no depende únicamente de unos pocos títulos. Oros géneros a destacar que muestran resultados similares son Disparos, Acción, Plataformas, y sobre todo, Rol. Este último alcanza un porcentaje superior al 30% en el año 2023, el año más reciente que se estudió en esta sección, lo cual lo convierte definitivamente en un género de interés al cual vale la pena investigar más. 
Los gráficos circulares nos permiten ver la evolución de la popularidad de los gpeneros a través de ciertos años de origen, pero también el cambio en la distribución de ventas: mientras que los juegos lanzados en 2008 parecen dividir de manera relativamente equitativa las ventas entre los géneros, el resto de años analizados siguen una tendencia diferente: uno o dos géneros dominan el gráfico, mientras que los demás tienen porcentajes con pequeñas diferencias entre ellos. Esto nos señala que en años posteriores fue donde se lanzaron más juegos con popularidad masiva que terminaron sesgando los resultados, algo que se determinó como común en la industria según los análisis previos.

Para el caso del segundo gráfico, se estudió la relación de los géneros con las regiones y no está de más reiterar que para este análisis se usaron únicamente el 80% de los datos, al igual que en el estudio enfocado principalmente en las ventas por región, por lo que las conclusiones que se saquen gracias a este subconjunto de los datos podrían ser imprecisas o inciertas. Dicho esto, el gráfico entrega información sobre Japón, que parece ser la única región que muestra tendencias diferentes a las del resto de regiones, mostrando un mayor interés por juegos de Rol (siendo este el género mas vendido en dicha región) y un menor interés por juegos de Disparos o Carreras.  

Como conclusión, este análisis adicional en los géneros permite dejar en claro una cosa: el género de Acción-Aventura es el más popular históricamente, pero eso no debe impedir que se ponga atención al resto de géneros relevantes, como Plataformas, Deportes o Disparos. También, algunos géneros pueden ser especialmente atractivos para ciertas regiones de clientes, como el género de Rol para los japoneses.


In [79]:
# 3 - Estudio de ventas por publisher

mode = lambda x: x.mode().iloc[0] if not x.mode().empty else None

sales_by_pub = df_sales_ind.groupby('publisher').agg(
    sales=('units_reported', 'sum'),
    games_released=('publisher', 'size'),
    sales_mean = ('units_reported', 'mean'),
    sales_median = ('units_reported', 'median'),
    origin=('year', 'min'),
).reset_index().sort_values('sales', ascending=False).copy()

top_20_publishers = sales_by_pub.head(15).copy()

# Gráfico ventas totales
fig = px.bar(top_20_publishers, x='publisher', y='sales')

fig.update_layout(
    title="Ventas por editora",
    xaxis_title="Editora",
    yaxis_title="Ventas"
)
fig.update_xaxes(tickangle=60) 
fig.update_traces(marker_color='crimson')

fig.show()

# Gráfico media/mediana


df_long = top_20_publishers.melt(id_vars ='publisher', value_vars=['sales_mean', 'sales_median'],
                              var_name='statistic', value_name='value')

fig = px.bar(top_20_publishers, x='publisher', y='sales_median')
fig.update_layout(
    title="Mediana de ventas por editora",
    xaxis_title="Editora",
    yaxis_title="Promedio/Mediana"
)
fig.update_xaxes(tickangle=60) 
fig.update_traces(marker_color='#19D3F3')
fig.show()

# Gráfico cantidad de juegos según año de origen

fig = px.bar(top_20_publishers, x='publisher', y='games_released')

fig.update_layout(
    title="Juegos publicados por editora",
    xaxis_title="Editora",
    yaxis_title="Ventas"
)
fig.update_xaxes(tickangle=60) 
fig.update_traces(marker_color='#00CC96')

fig.show()


# Gráfico subplot de géneros

genre_by_pub = df_sales_ind.groupby(
    ["publisher", "genre"],
    as_index=False
).agg({
    "units_reported": "sum",
}).sort_values('units_reported', ascending=False)

important_pubs = ["Nintendo", "Electronic Arts", "Rockstar Games", "Sony Interactive Entertainment"]

fig = make_subplots(
    rows=2,
    cols=2,
    specs=[
        [{"type": "pie"}, {"type": "pie"}],
        [{"type": "pie"}, {"type": "pie"}]
    ],
    subplot_titles=[
        "Géneros más vendidos - Nintendo", 
        "Géneros más vendidos - Electronic Arts",
        "Géneros más vendidos - Rockstar Games",
        "Géneros más vendidos - Sony Interactive"]
)

for i, pubs in enumerate(important_pubs):
    df_pubs = genre_by_pub[genre_by_pub['publisher'] == pubs].head(10).copy()
    fig.add_trace(
        go.Pie(labels=df_pubs["genre"], values=df_pubs["units_reported"]),
        row = i // 2 + 1,
        col = i % 2 + 1
    )
    fig.update_layout(
    width=1000,
    height=800
    )

for annotation in fig.layout.annotations:
    annotation.y += 0.04

fig.show()


# Gráfico años de origen editoras

genre_by_pub = genre_by_pub.drop_duplicates(subset=['publisher'], keep='first')
top_20_publishers['best_genre'] = top_20_publishers['publisher'].map(genre_by_pub.set_index('publisher')['genre'])

fig = px.scatter(top_20_publishers, x="origin", y='publisher',
                 size='sales', color='best_genre',
                 size_max=60)

fig.show()

### Ventas por editora  

Para esta parte de la investigación, se consideraron las 15 editoras con más ventas acumuladas como varaiables de interés. El promedio es una medida que da demasiado crédito a editoras que han publicado unos pocos juegos, mientras que ignora onfluyentes marcas que históricamente han producido bastantes éxitos, pero también bastantes juegos sin mucha relevancia. Por otro lado, la mediana es una estadística decente, pero que comete en menor medida el mismo error que el promedio, aunque es mucho más confiable para decidir las variables de interés que este último. Dicho esto, se decidió que era más adecuado utlizar una medida que tomara en cuenta el éxito total e histórico, ya que es un factor importante para la popularidad. Sin embargo, se utilizó la mediana como medida secundaria, análisis en el que profundizaremos al hablar del segundo gráfico.  

En este primer gráfico podemos observar que Nintendo es la editora con más ventas totales, superando con creces todas las demás. Le siguen Activision y Electronic Arts. En el gráfico podemos ver otras editoras responsables de públicar títulos exitosos, como "Mojang", responsable de "Minecraft", o "Rockstar Games", responsable de la saga "Grand Theft Auto".  

### Medianas en editoras exitosas  

Al analizar el gráfico con las medianas de las 15 editoras con más ventas totales, es notorio que el orden del tamaño de las barras no es el mismo que el del gráfico anterior. Empresas como "Rockstar Games", "Mojang" o "Sony Interactive Entertainment" tienen medianas especialmente altas, mientras que empresas con grandes cifras de ventas totales parecen tener medianas pequeñas en comparación, como "Activision" o "Ubisoft".

### Juegos publicados por editora  

Gracias a este tercer gráfico queda claro que las diferencias observadas en las medianas se deben principalmente a la cantidad de juegos publicados por editora. "Activision" y "Ubisoft" muestran menores medianas ya que son las empresas con mayor cantidad de juegos publicados. Mientras tanto, "Rockstar Games", "Mojang" y "Sony Interactive Entertainment" son editoras con menos juegos publicados y altas medianas. Exceptuando el caso de Mojang, que solo ha publicado 10 juegos en total, los resultados de mediana de "Rockstar Games" y "Sony Interactive Entertainment" son interesantes de analizar.  

### Géneros más vendidos de editoras de interés  

Para este estudio y el siguiente, se consideró la variable de géneros combinados con editoras. En este caso, se consideraron cuatro editoras como "interesantes" para analizar, combinando marcas que dominan en ventas totales y marcas que muestran considerables valores de mediana.  

Nintendo tiene como sus géneros más vendidos Plataformas, Rol y Deportes, géneros que ya se han considerado relevantes en previos análisis y que hacen sentido cuando consideramos las dos sagas más vendidas por Nintendo, "Super Mario." y "Pokemon".  

Electronic Arts muestra una distribución diferente, siendo sus géneros más vendidos Disparos, Simulación y Carreras, mostrando un lado diferente a lo investigado y abriendo la posibilidad a investigar géneros más a fondo para determinar si se trata de algo que funciona para Electronic Arts o si géneros como Simulación pueden tener éxito consistentemente.  

Finalmente, tenemos distribuciones curiosamente similares entre Rockstar Games y Sony Interactive Entertainment. Ambos muestran que sus géneros más vendidos son, en orden, Acción-Aventura, Acción y Aventura, lo cual marca una tendencia clara y consolida Acción, Aventura y su género compuesto como un componente relacionado con el éxito de ventas y como géneros a considerar para lanzar un videojuego exitoso. Estos géneros abarcan la mayoría de ventas de estas dos editoras de éxito, y también están presentes entre los mejores géneros de Nintendo y Electronic Arts, lo cual comunica de la alta tasa de éxito que tienen videojuegos que combinan géneros populares y editoras influyentes. Un dato importante a tener en cuenta es que la editora Sony Interactive Entertainment era antes llamada Sony Computer Entertainment, la cual está listada como una editora distinta en los gráficos analizados. Por esto, se podría argumentar que las ventas de ambas editoras se podrían combinar para considerar sus ventas totales acumuladas como una sola editora, pero se decidió tomar en cuenta la forma en la que están divididas en el conjunto de datos.

### Géneros, ventas y años de origen  

En el último gráfico se combina el total de ventas, el año de origen (en el que publicaron su primer juego), y el genero más vendido de cada editora. Gracias a esto, se comprueba que el año de origen, y por consecuencia, los años de actividad de la empresa editora, tienen estrecha relación con el número de ventas totales: Por norma general, las editoras más antiguas han acumulado más ventas que las más recientes.  

Otro factor importante a analizar son los géneros: Disparos y Plataformas son los géneros más vendidos de las editoras más antiguas. Esto no quiere decir que dichos géneros se vendieran más con juegos antiguos, si no que a lo largo de los años de actividad de grandes editoras como Nintendo y Activision, los juegos que más se han vendido suelen tener los géneros Plataformas y Disparos respectivamente. Se podría concluir a partir de aquello que estos géneros deben en cierta parte su éxito a la tradición y costumbre que grandes sagas de reconocidas editoras tienen al lanzar juegos, como es el caso de la saga "Super Mario" (Plataformas) de Nintendo, o la saga "Call of Duty" (Disparos) de Activision.  

Otro género que parece tener una tendencia marcada relacionada a los años es el de Rol, que aparece como el género más vendido en editoras con años de origen 2003 y 2008, años mucho más recientes que los años de origen de las editoras de las que hablamos en el párrafo anterior. Esto expresa un descubrimiento que coincide y por lo tanto confirma algo que se observó en el análisis de géneros relacionados a a ños de origen: a medida que avanzamos en años, el género de Rol se vuelve más vendido, siendo este el género más vendido para juegos con 2024 como su año de origen. Esta información es de suma importancia porque comunica del éxito de un género específicamente en la actualidad.  

Por último, es importante reconocer que nuevamente observamos resultados similares a otros estudios con respecto al género Acción-Aventura: a lo largo de todos los años se han originado editoras que tienen al género como su mejor vendido, lo cual consolida aún más la visión que se tenía de este género como un ingrediente que es consistentemente capaz de facilitar el éxito de un videojuego.


In [80]:
# 4 - Estudio de series y conclusión

top_10_series = df_sales_series.iloc[:10]
top_10_series = top_10_series.sort_values('total_shipped', ascending=False)

fig = px.bar(top_10_series, x='total_shipped', y='title',
             orientation='h', category_orders={'title': top_10_series['title'].tolist()}
)

fig.update_layout(
    title="Top 10 sagas exitosas de videojuegos",
    xaxis_title="Ventas",
    yaxis_title="Sagas"
)

fig.show()


fig = px.scatter(top_10_series, x="year", y='title',
                 size='total_shipped', color='genre',
                 size_max=60)

fig.show()

### Conjunto de datos de sagas

Como último análisis antes de las conclusiones del estudio, se utilizará el segundo conjunto de datos obtenido de la fuente original, correspondiente a sagas completas de videojuegos en lugar de entregas individuales. Existen razones importantes para revisar estos resultados, estas siendo:
+ El conjunto de datos de sagas contiene información relacionada a franquicias exitosas que no aparecen en el conjunto de datos individual, como Tetris o Pokemon. 
+ Permite analizar la industria de videojuegos desde un punto más amplio y enteder mejor lo que la caracteriza.
+ Permite comparar el rendimiento de las ventas individuales con las de las sagas, evidenciando de forma más clara lo que es exitoso y lo que no.

Separar ambos dataset era necesario para analizar correctamente las ventas de cada videojuego, pero provocó que ocurrieran circunstancias como esta, que hacen necesario el vistazo al conjunto de sagas. Sin embargo, solo se realizará un análisis general de las sagas, ya que el conjunto de datos de juegos individuales fue considerado el principal y el más completo para investigar los factores más específicos que se han estudiado previamente. Investigaciones externas al conjunto de datos fueron realizadas para complementar la informació de las franquicias exitosas, al no disponer del mismo nivel de especificidad y detalles que generan conclusiones del conjunto de datos individuales.  

En los gráficos, podemos observar que Tetris y Pokémon, las dos sagas cuyos juegos no están presentes en el conjunto de datos individual, son las dos sagas con mayor venta acumulada de todas. En sus videojuegos priman los géneros de Puzzle y Rol respectivamente, y ambas han superado las 500 millones de copias de juegos vendidas. Gracias al segundo gráfico podemos comprobar que ambas se originan antes del año 2000. En el segundo gráfico también es posible observar que no existe una tendencia fija de ventas o géneros relacionada a los años de origen: muchas franquicias han triunfado más que otras que se han originado antes. Tetris y Super Mario destacan por ser sagas con enormes cantidades de ventas acumuladas que se originaron antes de los años 90, y son los únicos ejemplos de sagas en el gráfico que tienen sus respectivos géneros como el más vendido. El género de plataformas ha sido determinado como un género de éxito en este estudio, pero el género Puzzle no mostró resultados destacables en ninguno de los estudios realizados, por lo que podemos atribuir a Tetris un caso único de éxito de este tipo de videojuegos. Después de realizar una investigación al respecto, se descubrió que la versión que disparó las ventas de la saga Tetris fue un juego lanzado para móviles en la década de los 2000, por la destacada y ya analizada editora "Electronic Arts", por lo que no podemos atribuir su éxito únicamente a la época en la que se originó la saga.

Pokemon y Super Mario son las dos franquicias más exitosas de la editora más popular, Nintendo. Han mantenido una popularidad constante hasta el día de hoy mediante personajes queridos y diseños atractivos, construyendo un ecosistema alrededor de los videojuegos, lo cual las volvió icónicas en el mundo de los videojuegos.   

El resto de franquicias han sido exploradas de alguna forma u de otra en los análisis previos, y casi todas comparten la característica de tener varios juegos exitosos que con parte de una línea de entregas principales de la saga que están relacionadas entre sí, a las cuales deben su éxito ya que es lo que hace a los jugadores volver a comprar cada nueva entrega. El único caso que no forma parte de esto es Minecraft, que es el nombre de la saga pero también del único juego realmente exitoso perteneciente a ella, con el mismo nombre. Un dato a añadir de este videojuego es que en este conjunto de datos su género aparece como "Sandbox", es decir, es un juego que destaca por la libertad que da a los jugadores de crear y explorar un mundo virtual, lo cual había sido destacado como una característica importante que puede llevar al éxito de un videojuego, y es la razón por la que Minecraft está catalogado como un juego de "Acción-Aventura" en el conjunto de datos princpal.



### Combinaciones exitosas

Habiendo concluido con el análisis del conjunto de datos, se puede empezar a responder la pregunta inicial de la investigación: ¿Qué elementos hacen a un videojuego exitoso? 
Para ello, es posible generar combinaciones a partir de las características estudiadas, con el afán de de proyectar ejemplos de lo que podrían ser videojuegos exitosos a partir de las conclusiones.  

Uno de los aspectos importantes a considerar de un videojuego es su género. Como se ha evidenciado en el análisis, hay muchos géneros que han tenido éxito en la industria, pero uno ha destacado como un género constantemente relacionado al éxito: Acción-Aventura. Al ser un género compuesto, implicaría que el videojuego tiene aspectos relacionados tanto a la Acción como a la Aventura, y el análisis ha probado que la presencia de Acción es la más relevante de las dos. Esto nos lleva a otro punto: la libertad del jugador como elemento indispensable de un videojuego de éxito. Esta se puede manifestar con un género compuesto como Acción-Aventura, pero también se asocia en géneral a videojuegos que combinan diferentes géneros para entregar una experiencia única al jugarlos.  

Una característica relacionada a la libertad de exploración puede manifestarse en otro ejemplo de juego exitoso: un mundo inmersivo combinado con una historia cautivadora. Este tipo de videojuegos combinan muy bien con el confiable género de Acción-Aventura, pero sobretodo con el género de Rol, el cual ha tomado mayor fuerza en años más recientes, según el análisis. Ejemplos de este tipo de juegos son los ya mencionados "The Witcher 3: Wild Hunt" y "The Elder Scrolls V: Skyrim". Otro enfoque de videojuegos que pueden tener éxito con este tipo de características es el de relacionarlo con franquicias icónicas. Un ejemplo de esto es el juego "Hogwarts Legacy", basado en el universo de Harry Potter, que está entre los más vendidos incluso habiendo sido lanzado hace menos de 4 años.

Otro elemento que ha surgido como un responsable de éxito han sido mecánicas de control innovadoras, como es el caso de Wii Sports, del año 2006, pero no es el único ejemplo: Guitar Hero (con 40 millones de ventas) es otra franquicia que innovó con los controles, sustituyendo el mando tradicional por una guitarra de plástico. Sin embargo, esta innovación mecánica no tiene por qué manifestarse en la infraestructura, si no que también puede ocurrir en el mismo videojuego. La idea es generar una experiencia nueva para el jugador, proponiendo formas de jugar a un juego que se destacan por lo diferentes e interactivas que son. Géneros que combinan bien con este tipo de juegos son Deportes y Fiesta.  

Una parte importante de muchos videojuegos es el multijugador: A medida que ha evolucionado la industria, han aparecido más y más juegos que proporcionan formas de interactuar con otros jugadores de diferentes formas. Existen videojuegos cuyo apartado principal es el multijugador, y otros que lo tienen como complemento al contenido primario. Para juegos donde el componente multijugador es algo central, géneros como Plataformas, Acción, Carreras e incluso Fiesta pueden ser buenas ideas para construir situaciones ideales para jugar con un grupo de amigos, como es el caso de "Human Fall Flat" o "Mario Kart 8 Deluxe". Otro enfoque de esta característica es el de videojuegos online donde se interactúa con jugadores del resto del mundo. Un ejemplo de esto es la modalidad "Battle Royale", la cual fue mencionada en uno de los análisis previos y es una buena aplicación del género Disparos. Otro ejemplo de esto son los juego de género MMO, o multijugador masivo, siendo la entrega más exitosa de este género "World of Warcraft", que destaca por su accesibilidad y expansiones constantes.

Por último, uno de los factores más importantes que han causado el éxito de videojuegos: iconicidad y representatividad. Muchos de los videojuegos que se han analizado como exitosos provienen de una franquicia icónica o pertenecen a una serie de entregas conocidas, en donde parte del éxito de cada videojuego en la serie se debe a que las personas conocían las entregas anteriores y querían jugar la nueva. Esto se evidencia con la cantidad de videojuegos con estas características que se han revisado. Algunos ejemplos son:  

+ "Grand Theft Auto V", con prededcesores populares como "Grand Theft Auto: Vice City" (17.5 millones vendidas) y "Grand Theft Auto IV" (25 millones vendidas).  
+ "Call of Duty: Black Ops 3", con previos juegos de la saga como "Call of Duty: Modern Warfare" (41 millones vendidas) y "Call of Duty: Black Ops II" (31 millones vendidas).
+ "Mario Kart 8 Deluxe", con predecesores como "Mario Kart Wii" (37 millones vendidas) y "Mario Kart 7" (19 millones vendidas).  

Este es el caso para muchos de los juegos exitosos del conjunto de datos analizado, y es una de las razones por las cuales las franquicias exitosas se mantienen relevantes con el tiempo. Crear juegos cuyos personajes o historias pertenezcan a una franquicia que ya es conocida y querida puede volverla icónica, generando un ecosistema de popularidad alrededor de los videojuegos pertenecientes a la saga. 

In [82]:
df_sales_ind['log_sales'] = np.log1p(df_sales_ind['units_reported'])

fig = px.histogram(df_sales_ind, x='log_sales', nbins=30)
fig.show()

fig = px.box(df_sales_ind, y='log_sales')
fig.show()

upper_whisker = (
    df_sales_ind['units_reported'].quantile(0.75)
    + 1.5 * (
        df_sales_ind['units_reported'].quantile(0.75)
        - df_sales_ind['units_reported'].quantile(0.25)
    )
)

outlier_percent = ((df_sales_ind['units_reported'] > upper_whisker).sum()) / len(df_sales_ind) * 100

print(f"El porcentaje de datos atípicos corresponde a {np.round(outlier_percent, decimals=2)}%")

El porcentaje de datos atípicos corresponde a 12.28%


### Conclusiones finales

Como se puede observar en los gráficos anteriores y se ha deducido y analizado durante todo los pasos, el conjunto de datos de ventas de videojuegos es uno muy sesgado por los éxitos, que contienen cifras altísimas en comparación a los números que venden normalmente los videojuegos. Hablamos de un conjunto cuya distribución contiene un 12.28 % de datos atípicos, los cuales podrían considerarse como exitosos. Teniendo en cuenta que solo un pequeño porcentaje de videojuegos alcanzan éxito en ventas, es muy probable que esta sea una hazaña difícil de lograr, que podría requerir años de planeación y desarrollo o incluso varios intentos. La industria d elos videojuegos esté en constante cambio y evolución, y no siempre es fácil comprender lo que resonará con los jugadores y lo que no. Este estudio intentó recolectar y analizar datos sobre lo que históricamente ha funcionado como exitoso, ya sea en las primeras épocas de la industria o en años recientes. Se ha explorado con diferentes géneros, y se ha investigado sobre títulos y editoras específicas que representan ejemplos de éxito. Tanto la sección anterior como esta apuntan a reunir las ideas captadas durante todo el análisis y hacer un resumen que pueda responder a la pregunta original de la investigación, y se llegó a conceptos claves como la libertad del jugador, la innovación y la representatividad.